src/main/java/org/desarrolladorslp/technovation/services/impl/MessageServiceImpl.java

```diff
@@ -34,33 +43,67 @@
 
     @Override
     @Transactional
-    public void markMessageAsRead(UUID messageId){
+    public MessageBodyDTO getSpecificMessageByUser(UUID messageId, UUID userReceiverId) {
+        Message m = messageRepository.getBodyOfMessageByUser(messageId);
```

Comment:
It is very important to use the user in getBodyOfMessageByUser so any user can't read a message that doesn't belongs to him/her